stackcollapse-phpspy.pl

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/perl -w
+#
+# stackcolllapse-phpspy.pl  collapse phpspy samples into single lines.
+#
+# Parses php samples generated by phpspy and outputs stacks as
+# single lines, with methods separated by semicolons, and then a space and an
+# occurrence count. For use with flamegraph.pl.
+#
+# USAGE: ./stackcollapse-phpspy.pl infile > outfile
+#
+# Example Input:
+#   ...
+#   0 sleep <internal>:-1
+#   1 aaa /home/mlauter/profiling/sample.php:5
+#   2 bbb /home/mlauter/profiling/sample.php:10
+#   3 <main> /home/mlauter/profiling/sample.php:25
+#   0 sleep <internal>:-1
+#   1 aaa /home/mlauter/profiling/sample.php:5
+#   2 <main> /home/mlauter/profiling/sample.php:28
+#   0 sleep <internal>:-1
+#   1 aaa /home/mlauter/profiling/sample.php:5
+#   2 bbb /home/mlauter/profiling/sample.php:10
+#   3 ccc /home/mlauter/profiling/sample.php:15
+#   4 <main> /home/mlauter/profiling/sample.php:22
+#   ...
+#
+# Example Output:
+#   <main>;ccc;bbb;aaa 1
+#   <main>;aaa 1
+#   <main>;bbb;aaa;sleep 1
+#
+# To make a flamegraph:
+# git clone https://github.com/brendangregg/FlameGraph flamegraph
+# ./stackcollapse-phpspy.pl infile | ./flamegraph/flamegraph.pl > svg.out
+
+use strict;
+use warnings;
+
+# might not need this config tbh
+use Getopt::Long qw(:config gnu_getopt no_ignore_case);
+
+# parameters
+my $help = 0;
+my $do_foo = 0;
+
+sub usage {
+    die <<USAGE_END;
+USAGE: $0 [options] infile > outfile\n
+    --h|help     # print this message
+USAGE_END
+}
+
+GetOptions(
+    'help|h' => \$help
+) or usage();
+usage() if $help;
+
+# internals
+my %stacks;
+my @frames;
+my $collapsed;
+
+sub remember_stack {
+    my ($stack, $count) = @_;
+    $stacks{$stack} += $count;
+}
+
+while (defined($_ = <>)) {
+    next if !m/^\d/;
+    chomp;
+
+    my @args = split ' ', $_;
+    my ($depth, $func) = @args[0,1];
```

Comment:
This line and the line above it can be collapsed (heh) into one. You should be able to do either something like this (using `undef` as a placeholder for the remaining element)

My ($depth, $func, under) = split...

Or you can take a slice from split itself like 

My ($depth, $func) = (split..)[0,1]